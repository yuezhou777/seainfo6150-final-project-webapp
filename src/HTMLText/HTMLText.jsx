import React from 'react';

const HTMLText = (props) => {

    return <div dangerouslySetInnerHTML={{ __html: props.content }} />;

}

export default HTMLText;