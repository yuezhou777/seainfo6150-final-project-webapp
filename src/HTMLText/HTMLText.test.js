import React from 'react'
import {render} from "@testing-library/react";
import HTMLText from "./HTMLText";

describe('HTMLText test', () => {

    it('render successfully ', () => {

        const text = '<p>hello</p><p>world</p>';

        const {container} = render(<HTMLText content={text} />);
        
        expect(container).toMatchSnapshot();
    });

})