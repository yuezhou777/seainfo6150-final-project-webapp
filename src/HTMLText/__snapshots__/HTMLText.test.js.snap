// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HTMLText test render successfully  1`] = `
<div>
  <div>
    <p>
      hello
    </p>
    <p>
      world
    </p>
  </div>
</div>
`;

exports[`html text test render successfully  1`] = `
<div>
  <div>
    <p>
      hello
    </p>
    <p>
      world
    </p>
  </div>
</div>
`;
